src/v2/providers/eventarc.ts

```diff
@@ -0,0 +1,163 @@
+  handler: CloudEventHandler
+): CloudFunction<CloudEvent<T>>;
+
+export function onCustomEventPublished<T = any>(
+  eventType: string,
+  optsOrHandler:
+    CloudEventHandler
+    | EventarcTriggerOptions,
+  handler?: CloudEventHandler,
+): CloudFunction<CloudEvent<T>> {
+  if (arguments.length < 2) {
+    throw new Error(
+      'Missing required parameters.'
+    );  
+  }
+  if (arguments.length == 2) {
+    if (typeof optsOrHandler !== 'function') { 
+      throw new Error(
+        'Expected second parameter to be the event handler function.'
+      );
+    }
+    handler = optsOrHandler as CloudEventHandler;
+    optsOrHandler = {} as EventarcTriggerOptions;
+  } else if (arguments.length == 3) {
+    optsOrHandler = optsOrHandler as EventarcTriggerOptions;
+  }
+
+  const opts = optsOrHandler as EventarcTriggerOptions;
+  const func = (raw: CloudEvent<unknown>) => {
+    return handler(raw as CloudEvent<T>);
+  };
+
+  func.run = handler;
+
+  const channel = opts.channel ?? 'locations/us-central1/channels/firebase';
+
+  Object.defineProperty(func, '__trigger', {
+    get: () => {
+      const baseOpts = options.optionsToTriggerAnnotations(
+        options.getGlobalOptions()
+      );
+      const specificOpts = options.optionsToTriggerAnnotations(opts);
+      const trigger = {
+        platform: 'gcfv2',
+        ...baseOpts,
+        ...specificOpts,
+        labels: {
+          ...baseOpts?.labels,
+          ...specificOpts?.labels,
+        },
+        eventTrigger: {
+          eventType,
+          filters: opts.filters,
+          channel,
+        },
+      };
+      if (!opts.filters) {
+        delete trigger.eventTrigger.filters;
+      }
+
+      return trigger;
+    },
+  });
+
+  // TypeScript doesn't recognize defineProperty as adding a property and complains
+  // that __endpoint doesn't exist. We can either cast to any and lose all type safety
+  // or we can just assign a meaningless value before calling defineProperty.
+  func.__endpoint = {} as ManifestEndpoint;
+
+  // SDK may attempt to read FIREBASE_CONFIG env var to fetch the default bucket name.
+  // To prevent runtime errors when FIREBASE_CONFIG env var is missing, we use getters.
+  Object.defineProperty(func, '__endpoint', {
+    get: () => {
+      const baseOpts = options.optionsToEndpoint(options.getGlobalOptions());
+      const specificOpts = options.optionsToEndpoint(opts);
+
+      const endpoint: ManifestEndpoint = {
+        platform: 'gcfv2',
+        ...baseOpts,
+        ...specificOpts,
+        labels: {
+          ...baseOpts?.labels,
+          ...specificOpts?.labels,
+        },
+        eventTrigger: {
+          eventType,
+          eventFilters: opts.filters,
+          retry: false,
+          channel,
+        },
+      };
+      if (!opts.filters) {
+        delete endpoint.eventTrigger.eventFilters;
+      }
+      copyIfPresent(endpoint.eventTrigger, opts, 'retry', 'retry');
+      return endpoint;
+    },
+  });
```

Comment:
As noted in the comment, we have this "trick" of lazily defining `__endpoint` property b/c it some trigger definitions relies on existence of some environment variable injected by the Firebase CLI.

I don't think this is the case for custom event triggers (or of any v2 triggers!) - so feel free to just directly attach an `__endpoint` definition, e.g.:

https://github.com/firebase/firebase-functions/blob/270605dcc884933a7356406ef504ccd7a54cfaca/src/v2/providers/https.ts#L258-L267

Reply:
Done.